main/test/ca/uwaterloo/flix/language/phase/TestResolver.scala

```diff
@@ -106,7 +106,7 @@ class TestResolver extends AnyFunSuite with TestUtils {
     expectError[ResolutionError.InaccessibleStruct](result)
   }
 
-  ignore("InaccessibleStruct.02") {
+  test("InaccessibleStruct.02") {
```

Comment:
I'd like another test where the struct is only used in an expression.

Reply:
Why would this be an error if a struct is used in an expression?

Reply:
I phrased that poorly.

what I mean is I'd like to see something like:

```scala
def foo(): Int32 = {
  let _ = new S[r] { };
  123
}
```

where S doesn't exist.

So we have to detect the error in the expression, not just in the type signature.

Reply:
Got it

Sorry if I'm being dumb, but how is this different from "ResolutionError.NonExistentStruct.01"?

Reply:
I am the dumb one. :P
I meant where S is inaccessible instead of "doesn't exist"

Reply:
Haha adding this test revealed a bug that opened a bit of a can of worms. I'll figure out how to push the fix but it spans a number of files.